wandb/sdk/artifacts/artifact.py

```diff
@@ -1165,12 +1168,15 @@ def add_file(
             file_name_parts[0] = b64_to_hex_id(digest)[:20]
             name = os.path.join(file_path, ".".join(file_name_parts))
 
-        return self._add_local_file(name, local_path, digest=digest, policy=policy)
+        return self._add_local_file(
+            name, local_path, digest=digest, skip_cache=skip_cache, policy=policy
+        )
 
     def add_dir(
         self,
         local_path: str,
         name: Optional[str] = None,
+        skip_cache: Optional[bool] = False,
         policy: Optional[Literal["mutable", "immutable"]] = "mutable",
```

Comment:
is any of these actually Optional type?

Reply:
fixed the case for `policy` in the parent PR. In the case of `skip_cache`, it can accept the `None` value. The `ArtifactManifestEntry` class takes care of setting the correct value if it receives `None` as input...

Reply:
https://github.com/wandb/wandb/pull/7098/files#diff-44409c7941af2a88067689e280ec19721100a5ad1d770d64c248434121eeecd3